[New Sample] Azure Open AI based sample "Enterprise Search App"

ChetanSharma-msft opened the pull request:
Open AI based sample with below capabilities:

1) Bot File Upload to Blob
2) Open AI Embeddings
3) Redis Search
4) Open AI Completion API

ChetanSharma-msft commented on samples/bot-ai-enterprise-search/nodejs/README.md:
```diff
@@ -0,0 +1,240 @@
+---
+page_type: sample
```
Need to add build pipeline and Link to root readme file.

ChetanSharma-msft replied:
Added build pipeline and root readme links.

ChetanSharma-msft reviewed (commented):
Added review comment.

ChetanSharma-msft commented on samples/bot-ai-enterprise-search/nodejs/prompts/chat/skprompt.txt:
```diff
@@ -0,0 +1,4 @@
+The following is a conversation with an AI assistant, its name is Teams Chef. 
```
Need to check and remove if it is not needed.

ChetanSharma-msft replied:
Removed it.

ChetanSharma-msft commented:
Executed build pipeline: https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5710770366

aratigurav reviewed (commented):
Please update the title to reflect Azure Open AI based sample. I can see some of the meeting app related information which is not relevant. In the deployment or how to run application mention about the valid domain to be added.

ChetanSharma-msft commented:
> Please update the title to reflect Azure Open AI based sample. I can see some of the meeting app related information which is not relevant. In the deployment or how to run application mention about the valid domain to be added.

1) Updated Title and other docs.
2) Removed meeting related fix and raised another PR for that.
3) Added step to update valid domain and configuration for web app.

aratigurav commented:
@ChetanSharma-msft couple of observations: 

1. Solution architecture diagram is not visible on readme page.
2. Interaction with bot -admin section is empty.
3. Interaction with bot- user section is empty.

ChetanSharma-msft commented:
Executed build pipeline: https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5729965655

Mohammed-MSFT reviewed (approved):
Changes Looks Fine Approving It.

ChetanSharma-msft commented:
> @ChetanSharma-msft couple of observations:
> 
> 1. Solution architecture diagram is not visible on readme page.
> 2. Interaction with bot -admin section is empty.
> 3. Interaction with bot- user section is empty.



> @ChetanSharma-msft couple of observations:
> 
> 1. Solution architecture diagram is not visible on readme page.
> 2. Interaction with bot -admin section is empty.
> 3. Interaction with bot- user section is empty.

Fixed the images and gif images.
